datafusion/optimizer/src/common_subexpr_eliminate.rs

```diff
@@ -622,18 +657,21 @@ impl CommonSubexprEliminate {
     ) -> Result<Transformed<(Vec<Expr>, LogicalPlan)>> {
         // Extract common sub-expressions from the expressions.
         self.find_common_exprs(vec![exprs], config, ExprMask::Normal)?
-            .map_data(|(mut new_exprs_list, common)| {
-                let new_exprs = new_exprs_list.pop().unwrap();
-
-                match common {
-                    // If there are common sub-expressions, then insert a projection node
-                    // with the common expressions between the original node and the
-                    // original input.
-                    Some((common_exprs, _)) => {
-                        build_common_expr_project_plan(input, common_exprs)
-                            .map(|new_input| (new_exprs, new_input))
-                    }
-                    None => Ok((new_exprs, input)),
+            .map_data(|common| match common {
```

Comment:
This isn't quite as concise as the previous version but I do think it is more explicit and easier to follow